index.d.ts

```diff
@@ -339,6 +339,32 @@ export class TimeoutError extends Error {
 	constructor();
 }
 
+/**
+A `Symbol` that can be returned by a `beforeRetry` hook to stop the retry.
+This will also short circuit the remaining `beforeRetry` hooks.
+
+@example
+```
+import ky, {stop as kyStop} from 'ky';
+
+(async () => {
+	await ky('https://example.com', {
+		hooks: {
+			beforeRetry: [
+				async (request, options, errors, retryCount) => {
+					const shouldStopRetry = await ky('https://example.com/api');
+					if (shouldStopRetry) {
+						return kyStop;
```

Comment:
This mechanism could be also applied to Got :)

Reply:
Yup. Let’s try it out here first and see how it works out.